packages/apify/src/actor.ts

```diff
@@ -1304,9 +1366,15 @@ export class Actor<Data extends Dictionary = Dictionary> {
      *
      * @param item Object or array of objects containing data to be stored in the default dataset.
      * The objects must be serializable to JSON and the JSON representation of each object must be smaller than 9MB.
+     * @param eventName If provided, the method will attempt to charge for the event for each pushed item.
      */
-    static async pushData<Data extends Dictionary = Dictionary>(item: Data | Data[]): Promise<void> {
-        return Actor.getDefaultInstance().pushData(item);
+    static async pushData<Data extends Dictionary = Dictionary>(item: Data | Data[]): Promise<void>;
+    static async pushData<Data extends Dictionary = Dictionary>(item: Data | Data[], eventName: string): Promise<ChargeResult>;
+    static async pushData<Data extends Dictionary = Dictionary>(item: Data | Data[], eventName?: string): Promise<ChargeResult | void> {
+        if (eventName === undefined) {
+            return await Actor.getDefaultInstance().pushData(item);
+        }
+        return await Actor.getDefaultInstance().pushData(item, eventName);
```

Comment:
why the `await`s? and why not just `pushData(item, eventName)` for both cases? its effectively the same, no? missing param will give you `undefined`, this will only differ when you use `arguments` property

---

right, because of the overloads. cant help myself, but i don't like overloads at all. if you want to have them here, you need to document all of them separately, otherwise only the first one will be properly documented in our API docs

i would consider a new method for this instead of overloads, but then again, I am not sure if we want to pollute this `Actor` class more, it always has tons of methods

Reply:
Sure, I can document those overloads separately. In this case, adding a parameter to `pushData` seemed like a less invasive solution to me, but a `pushDataAndCharge` method is also an option :shrug: